pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -162,6 +163,8 @@ public class MutableSegmentImpl implements MutableSegment {
   private RealtimeLuceneIndexRefreshState.RealtimeLuceneReaders _realtimeLuceneReaders;
 
   private final UpsertConfig.Mode _upsertMode;
+  private final boolean _enableDeleteInUpsert;
```

Comment:
Consider removing this config, and enable deletion whenever `_upsertDeleteKey` is configured